crates/astria-core/src/primitive/v1/mod.rs

```diff
@@ -488,6 +488,103 @@ where
     tree
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize))]
+#[cfg_attr(feature = "serde", serde(transparent))]
+pub struct TransactionId {
+    hash: String,
```

Comment:
There is no need to make this a `String`. Keep this as a `[u8; 32]` and decide in the `Display` implementation what this should format to.

Reply:
```suggestion
    inner: [u8; 32],
```